File: Algorithms/Strings/Two Strings.php
<?php

/*
 * Complete the 'twoStrings' function below.
 *
 * The function is expected to return a STRING.
 * The function accepts following parameters:
 *  1. STRING s1
 *  2. STRING s2
 */

function twoStrings($s1, $s2) {
    # Write your code here
    $len1 = strlen($s1);
    $len2 = strlen($s2);
    $count = 0;
    $hash = array_fill(0, 26, 0);
    for ($i = 0; $i < $len2; $i++) {
        $hash[ord($s2[$i]) - ord('a')] = 1;
    }
    for ($i = 0; $i < $len1; $i++) {
        if ($hash[ord($s1[$i]) - ord('a')] == 1) {
            $count++;
        }
    }
    return $count > 0 ? "YES" : "NO";
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$q = intval(trim(fgets(STDIN)));

for ($q_itr = 0; $q_itr < $q; $q_itr++) {
    $s1 = rtrim(fgets(STDIN), "\r\n");
    $s2 = rtrim(fgets(STDIN), "\r\n");

    $result = twoStrings($s1, $s2);

    fwrite($fptr, $result . "\n");
}

fclose($fptr);

File: Algorithms/Strings/Two Strings.py
import os

#
# Complete the 'twoStrings' function below.
#
# The function is expected to return a STRING.
# The function accepts following parameters:
#  1. STRING s1
#  2. STRING s2
#

def twoStrings(s1, s2):
    # Write your code here
    s1_chars = set(s1)
    s2_chars = set(s2)

    if s1_chars.intersection(s2_chars):
        return "YES"
    else:
        return "NO"


if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    q = int(input().strip())

    for q_itr in range(q):
        s1 = input()
        s2 = input()

        result = twoStrings(s1, s2)
        fptr.write(result + '\n')

    fptr.close()
